### PR TITLE
EVG-17549: Use dropdown for project org selection

### PR DESCRIPTION
**Pull request by sophstad**
EVG-17549

### Description

- Update Create Project and Move Project modals to use dropdown for selecting which GitHub org should be used.

### Screenshots

image

image

### CODE REVIEW

File: src/pages/projectSettings/CreateProjectModal.test.tsx [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
+import { selectLGOption } from "test_utils/utils";
 import { CreateProjectModal } from "./CreateProjectModal";
 
-const defaultOwner = "existing_owner";
-const defaultRepo = "existing_repo";
+const mockedUseNavigate = jest.fn();
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useNavigate: () => mockedUseNavigate,
+}));
```
Comment (khelif96) [inline src/pages/projectSettings/CreateProjectModal.test.tsx:18]: If the goal of this mock was to assert that the modal redirected the user you can test this without mocking useNavigate. 
`renderWithRouterMatch` exposes a history object that you can use to assert on the current location. 


https://github.com/evergreen-ci/spruce/blob/620562260a9e06b7996671a45e9ab33f9f2399cf/src/components/FilterBadges/useFilterBadgeQueryParams.test.tsx#L62-L75

File: src/pages/projectSettings/CreateProjectModal.tsx [hunk 2]
```diff
@@ -38,6 +41,11 @@ export const CreateProjectModal: React.VFC<Props> = ({
   });
   const [hasError, setHasError] = useState(true);
 
+  const { data } = useQuery<GetGithubOrgsQuery>(GET_GITHUB_ORGS);
```
Comment (khelif96) [inline src/pages/projectSettings/CreateProjectModal.tsx:44]: adding a `{skip: !open }` to the params would prevent this query from being kicked off if the user never opens this modal.

**Review by khelif96 — CHANGES_REQUESTED** [review]
Can we add/augment some cypress tests for this?

File: src/pages/projectSettings/CreateProjectModal.test.tsx (hunk 1 above)
Comment (sophstad) [reply src/pages/projectSettings/CreateProjectModal.test.tsx:19]: My goal here was really to remove a warning:
```
console.warn
    No routes matched location "/project/new-project-name/settings"
```
We've been kinda lax w.r.t. unit test warnings thus far but I think removing them would be preferable. Using the `renderWithRouterMatch` approach doesn't have a way to add additional routes at the moment, so I went with the mock instead.

Comment (khelif96) [reply src/pages/projectSettings/CreateProjectModal.test.tsx:19]: Oh great catch you should be able to augment `renderWithRouterMatch` to support that adding the following line to the method eliminates that warning and is reusable for all tasks.
image

**Review by khelif96 — APPROVED** [review]
LGTM

**Pull request merged**